src/plugins/intel_gpu/src/graph/graph_optimizer/prepare_buffer_fusing.cpp

```diff
@@ -844,6 +844,8 @@ void prepare_buffer_fusing::run(program& p) {
                                                                                             crop_params->input_offsets[0],
                                                                                             node.get_primitive()->axis,
                                                                                             false);
+                if (static_cast<bool>(crop_layout.data_padding) && node.get_users().front()->is_type<reorder>())
```

Comment:
As offline discussion, returning here is dangerous. So this condition must already be done within match function.